ncolony/ctllib.py

```diff
@@ -151,14 +159,36 @@ def _parseJSON(fname):
 _remove_parser.add_argument('name')
 _remove_parser.set_defaults(func=remove)
 _add_parser = _subparsers.add_parser('add')
-_add_parser.add_argument('name')
-_add_parser.add_argument('--cmd', required=True)
-_add_parser.add_argument('--arg', dest='args', action='append')
-_add_parser.add_argument('--env', action='append')
-_add_parser.add_argument('--uid', type=int)
-_add_parser.add_argument('--gid', type=int)
-_add_parser.add_argument('--extras', type=_parseJSON)
-_add_parser.add_argument('--env-inherit', dest='env_inherit', action='append')
+_add_parser.add_argument('name', help='The name of this command.')
+_add_parser.add_argument('--cmd',
+                         help='The path of the command to run.'
+                         ' This is mutually exclusive with cmd_and_args,'
+                         ' but one must be present.')
+_add_parser.add_argument('--arg', dest='args', action='append',
+                         help='An argument to the command.'
+                         ' Can be specified multiple times.'
+                         ' This is mutually exclusive with cmd_and_args.')
+_add_parser.add_argument('--env', action='append',
+                         help='A environment variable of the form'
+                         '`NAME=VALUE`.'
+                         ' Can be specified multiple times.')
+_add_parser.add_argument('--uid', type=int,
+                         help='The numeric user ID under which this command'
+                         ' will run.')
+_add_parser.add_argument('--gid', type=int,
+                         help='The numeric group ID under which this command'
+                         ' will run.')
+_add_parser.add_argument('--extras', type=_parseJSON,
+                         help='Additional data to store with this command.')
+_add_parser.add_argument('--env-inherit', dest='env_inherit', action='append',
+                         help='The name of environment variable that will be'
+                         ' inherited from the monitor process.'
+                         ' Can be specified multiple times.')
+_add_parser.add_argument("_cmd_and_args", nargs="*", metavar="cmd_and_args",
```

Comment:
Why not call this argument "cmd_and_args"? This would get rid of the warning and I don't see what would be lost -- the parser interface is internal anyway...

If there is a good reason, please add the lint pragma + explanation of the reason and merge.